### ISSUE TITLE
Undesired behavior with --format

### ISSUE THREAD

**Issue by ornithocoder** [realm/SwiftLint#2374]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

As `closure_end_indentation`'s description states, a closure end should have the same indentation as the line that started it, but this behavior conflicts with Xcode's indentation (^i) and SwiftLint's `--format`.

Below, a sample `.swiftlint.yml` file with the rule enabled, a swift file - `file.swift` - with one of the examples from `swiftlint rules closure_end_indentation` and the output, showing that the code is correct (closure ending with correct indentation, as imposed by the rule).

```bash
$ cat .swiftlint.yml
opt_in_rules:
  - closure_end_indentation

$ cat file.swift
SignalProducer(values: [1, 2, 3])
    .startWithNext { number in
        print(number)
    }

$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'file.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
```

However, when the command `swiftlint autocorrect --format` runs, it autocorrects the file and reformats it, reverting the autocorrection applied by the rule.

```bash
$ swiftlint autocorrect --format
Loading configuration from '.swiftlint.yml'
Correcting Swift files at paths 
Correcting 'file.swift' (1/1)
Done correcting 1 file!

$ cat file.swift
SignalProducer(values: [1, 2, 3])
    .startWithNext { number in
        print(number)
}
```

As result, `autocorrect` has to run twice - first to format and then to correct the files -, adding a undesired delay to the build phase:

```bash
$ swiftlint autocorrect --format
$ swiftlint autocorrect
```

Ideally `swiftlint autocorrect --format` should first format and then autocorrect the files.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `master`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? `swift build`
* Paste your configuration file:

```yml
opt_in_rules:
  - closure_end_indentation
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? no.
* Which Xcode version are you using (check `xcode-select -p`)? 9.4.1
